quasar-core/src/main/java/co/paralleluniverse/common/util/ExtendedStackTrace.java

```diff
@@ -123,14 +129,11 @@ public void visitEnd() {
                     }
                 });
 
-                if (descriptor.get() != null) {
-                    final String desc = descriptor.get();
-                    for (Member m : ms) {
-                        if (este.getMethodName().equals(getName(m)) && desc.equals(getDescriptor(m))) {
-                            method = m;
-                            break;
-                        }
-                    }
+                if (exactMatch.get() != null){
+                    method = getMatchingMethod(ms, methodName, exactMatch.get());
+                }
+                else if (descriptor.get() != null) {
+                    method = getMatchingMethod(ms, methodName, descriptor.get());
```

Comment:
More confusing uses of `AtomicReference.get()`... :confused:

Reply:
I'll fetch the value once into a local variable